contracts/exchange/RoyaltyManager.sol

```diff
@@ -139,8 +139,8 @@ contract RoyaltyManager is IRoyaltyManager, ManagerBase {
     }
 
     function buyOrderRoyalties(
-        LibOrder.AssetData calldata _asset,
-        uint256[] memory amountPerOrder
+        LibOrder.AssetData memory _asset,
```

Comment:
why did you flip this instead of just making both 'calldata'? I'm curious

Reply:
Surprisingly, switching the storage of the AssetData object to memory was cheaper for both of these functions.

Reply:
Interesting, maybe it's cheaper to use "memory" for structures? Can you check other places where we use 'calldata' for structures like LibOrder.AssetData and see if changing it to memory makes that function call cheaper?

Reply:
Yeah I checked it for all of them, it was sometimes cheaper and sometimes more expensive for structures to use memory. All of them already use whichever one's cheaper.

Reply:
If that's the case, we should just be consistent and use calldata for all function parameters instead of memory even if memory is slightly cheaper. It shouldn't be too much more expensive, and I think the benefits to consistency is much better than a very slight gas savings.